cli/code_scanner.py

```diff
@@ -210,13 +209,36 @@ def validate_zip_file_size(scan_type, zip_file_size):
 
 def perform_scan(cycode_client, zipped_documents: InMemoryZip, scan_type: str, scan_id: UUID, is_git_diff: bool,
                  is_commit_range: bool, scan_parameters: dict):
-    scan_result = cycode_client.commit_range_zipped_file_scan(scan_type, zipped_documents, scan_id) \
-        if is_commit_range else cycode_client.zipped_file_scan(scan_type, zipped_documents, scan_id, scan_parameters,
-                                                               is_git_diff)
+    if scan_type == SCA_SCAN_TYPE:
+        scan_result = perform_scan_with_polling(cycode_client, zipped_documents, scan_type, scan_parameters)
+    else:
+        scan_result = cycode_client.commit_range_zipped_file_scan(scan_type, zipped_documents, scan_id) \
+            if is_commit_range else cycode_client.zipped_file_scan(scan_type, zipped_documents, scan_id,
+                                                                   scan_parameters, is_git_diff)
 
     return scan_result
 
 
+def perform_scan_with_polling(cycode_client, zipped_documents: InMemoryZip, scan_type: str,
+                              scan_parameters: dict) -> ZippedFileScanResult:
+    # scan_async_result = cycode_client.zipped_file_scan_async(zipped_documents, scan_type, scan_parameters)
+    scan_async_result = ScanInitializationResponse(scan_id="d789d12c-194e-47c1-9fdd-5f8a1d10809d")
+    logger.debug("scan request has been triggered successfully, scan id: %s", scan_async_result.scan_id)
+    polling_timeout = configuration_manager.get_scan_polling_timeout_in_seconds()
+
+    end_polling_time = time.time() + polling_timeout
+    while time.time() < end_polling_time:
+        logger.debug("scan in progress")
+        scan_details = cycode_client.get_scan_details(scan_async_result.scan_id)
+        if scan_details.scan_status == SCAN_STATUS_COMPLETED:
+            return _get_scan_result(cycode_client, scan_async_result, scan_details)
+        if scan_details.scan_status == SCAN_STATUS_ERROR:
+            raise CycodeInternalError('error occurred while trying to scan zip file')
```

Comment:
log the scan error message

Reply:
i added the message so it will be sent in the report but I dont see why we should log to user stdout internal cycode errors